clients/src/main/java/org/apache/kafka/clients/producer/Partitioner.java

```diff
@@ -34,10 +34,9 @@
      * @param key The key to partition on (or null if no key)
      * @param keyBytes The serialized key to partition on( or null if no key)
      * @param value The value to partition on or null
-     * @param valueBytes The serialized value to partition on or null
      * @param cluster The current cluster metadata
      */
-    public int partition(String topic, Object key, byte[] keyBytes, Object value, byte[] valueBytes, Cluster cluster);
```

Comment:
This is a public interface, so we can't change the parameters without a KIP. We need to consider compatibility with others who may be using the old interface. This is likely why this parameter still exists.  See https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Improvement+Proposals.

Reply:
I can pass a dummy byte array to the partitioner instead I guess